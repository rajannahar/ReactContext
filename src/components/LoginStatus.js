import React, {useContext} from 'react'
import { AuthContext } from '../AuthContext/AuthContext'

const LoginStatus = () => {

  const status = useContext(AuthContext)
  const {isLoggedIn} = status

  return(
    <>
      <h2>Login status: {isLoggedIn.toString() === "true" ? 'Logged in' : 'Logged out'}</h2>
    </>
  )
}

export default LoginStatus
