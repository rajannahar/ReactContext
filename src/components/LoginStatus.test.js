import React from 'react';
import { render } from '@testing-library/react'
import { getByTestId } from '@testing-library/dom'
import LoginStatus from './LoginStatus';

test('should render login status', () => {
  const { getByTestId } = render(<LoginStatus />)
  expect(getByTestId('loginStatus')).toBeTruthy()
})

test('should have text "Login status: Logged out"', () => {
  const { getByTestId } = render(<LoginStatus />)
  const heading = getByTestId('loginStatus')
  expect(heading.innerHTML).toEqual('Login status: Logged out')
})

it("matches snapshot", () => {
  const { asFragment } = render(<LoginStatus />);
  expect(asFragment()).toMatchSnapshot();
});
