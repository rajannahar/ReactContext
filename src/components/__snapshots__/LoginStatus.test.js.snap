// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2
    data-testid="loginStatus"
  >
    Login status: Logged out
  </h2>
</DocumentFragment>
`;
